client/app/bundles/course/lesson-plan/containers/MilestoneFormDialog/MilestoneForm.jsx

```diff
@@ -1,63 +1,100 @@
 import PropTypes from 'prop-types';
 import { FormattedMessage } from 'react-intl';
-import { reduxForm, Field, Form } from 'redux-form';
-import renderTextField from 'lib/components/redux-form/TextField';
-import RichTextField from 'lib/components/redux-form/RichTextField';
+import { Controller, useForm } from 'react-hook-form';
+import * as yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import FormDateTimePickerField from 'lib/components/form/fields/DateTimePickerField';
+import FormRichTextField from 'lib/components/form/fields/RichTextField';
+import FormTextField from 'lib/components/form/fields/TextField';
+import ErrorText from 'lib/components/ErrorText';
 import formTranslations from 'lib/translations/form';
-import DateTimePicker from 'lib/components/redux-form/DateTimePicker';
 import translations from 'course/lesson-plan/translations';
-import { formNames, fields } from 'course/lesson-plan/constants';
+import { fields } from 'course/lesson-plan/constants';
 
 const { TITLE, DESCRIPTION, START_AT } = fields;
 
-const validate = (values) => {
-  const errors = {};
+const validationSchema = yup.object({
+  title: yup.string().required(formTranslations.required),
+  description: yup.string().nullable(),
+  start_at: yup.date().nullable().required(formTranslations.required),
```

Comment:
Hmm how does `nullable` and `required` work together?

Reply:
The purpose of combining both nullable and required is to prevent the following error when the form is submitted with empty field: start_at must be a `date` type, but the final value was: `Invalid Date`. 

When a new form is created, the field contains empty string. So we want to force the field to return the required error instead by bypassing the type check.